cirq-core/cirq/value/condition.py

```diff
@@ -107,17 +110,20 @@ def resolve(
     ) -> bool:
         if self.key not in classical_data.keys():
             raise ValueError(f'Measurement key {self.key} missing when testing classical control')
-        return classical_data.get_int(self.key) != 0
+        return classical_data.get_int(self.key, self.index) != 0
 
     def _json_dict_(self):
-        return json_serialization.dataclass_json_dict(self)
+        fields = ['key'] if self.index == -1 else ['key', 'index']
```

Comment:
If `Condition` has been in a released version of Cirq, this is needed for backwards compatibility (e.g. deserializing old `Condition`s that don't have an `index`).

If not, we can just always serialize both fields.